shoes-swt/lib/shoes/swt/common/image_handling.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+class Shoes
+  module Swt
+    module Common
+      module ImageHandling
+        # Why copy the file to a temporary location just to pass a different name
+        # to load? Because SWT doesn't like us when we're packaged!
+        #
+        # Apparently the warbler-style path names we end up with for relative
+        # image paths don't cross nicely to SWT, so we need to resolve the paths
+        # in Ruby-land before handing it over.
+        def load_file_image_data(name)
+          tmpname = File.join(Dir.tmpdir, "__shoes4_#{Time.now.to_f}_#{File.basename(name)}")
```

Comment:
Ported over from prior code, but I don't see any reason for `to_f` vs `to_i`. Updating.